backend/dataall/modules/redshift_datasets_shares/services/redshift_table_share_processor.py

```diff
@@ -18,11 +33,477 @@ def __init__(self, session, share_data, shareable_items, reapply=False):
+                    f'tables {[t.name for t in self.tables]} '
+                    f'from source {self.source_connection.name} in namespace {self.source_connection.nameSpaceId}'
+                    f'with target {self.target_connection.name} in namespace {self.target_connection.nameSpaceId}'
+                    f'due to: {e}'
+                )
+                if not self.reapply:
+                    new_state = revoked_item_SM.run_transition(ShareItemActions.Failure.value)
+                    revoked_item_SM.update_state(self.session, self.share.shareUri, new_state)
+                else:
+                    for table in self.tables:
+                        share_item = ShareObjectRepository.find_sharable_item(
+                            self.session, self.share.shareUri, table.rsTableUri
+                        )
+                        ShareStatusRepository.update_share_item_health_status(
+                            self.session, share_item, ShareItemHealthStatus.Unhealthy.value, str(e), datetime.now()
+                        )
+                return False
+            return success
 
     def verify_shares(self) -> bool:
+        """
+        1) (in source namespace) Check the datashare exists
+        2) (in source namespace) Check that schema is added to datashare
+        3) (in source namespace) Check the access is granted to the consumer cluster to the datashare
+        4) (in target namespace) Check that local db exists
+        5) (in target namespace) Check that the redshift role has access to the local db
+        6) (in target namespace) Check that external schema exists
+        7) (in target namespace) Check that the redshift role has access to the extenal schema
+        For each table:
+            8) (in source namespace) Check that table is added to datashare
+            9) (in target namespace) Check that the redshift role has select access to the requested table in the local db.
+            10) (in target namespace) Check that the redshift role has select access to the requested table in the external schema.
+        """
+
+        log.info('##### Verifying Redshift tables #######')
+        if not self.tables:
+            log.info('No tables to verify. Skipping...')
+        else:
+            tbl_level_errors = []
+            ds_level_errors = []
+            redshift_client_in_source = redshift_share_data_client(
+                account_id=self.share_data.source_environment.AwsAccountId,
+                region=self.share_data.source_environment.region,
+                connection=self.source_connection,
+            )
+            redshift_client_in_target = redshift_share_data_client(
+                account_id=self.share_data.target_environment.AwsAccountId,
+                region=self.share_data.target_environment.region,
+                connection=self.target_connection,
+            )
```

Comment:
nit: can move init of source and target clients to class `init()` if is called on every method?

Reply:
local_db and external_schema absolutely yes.  As for the redshift_clients, I am not so sure. I do not want to introduce tightly coupled classes in our code, at least not in the constructor; the risk is to end up having the situation of the lf_share_managers. I can introduce a function to reduce the code duplication, let me know if that is ok